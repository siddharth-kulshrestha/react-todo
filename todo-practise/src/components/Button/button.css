.button-container {
    text-align: center;
    border-width: thin;
    border-style: solid;
    border-radius: 5%;
    border-color: black;
}