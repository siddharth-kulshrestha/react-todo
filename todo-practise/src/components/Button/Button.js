import React from 'react';
import './button.css';

export const Button = ({ onClick, text }) => (
    <span className="button-container" onClick={onClick} >
        {text}
    </span>
);
