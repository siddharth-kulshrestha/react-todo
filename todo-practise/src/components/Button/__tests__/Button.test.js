import React from 'react';
import { shallow, configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import { Button } from '../Button';

configure({adapter: new Adapter()});

// Snapshot test for Button component
describe('Button ', () => {
    it('should render correctly ', () => {
        const component = shallow(<Button text={'hello world'} />);
        expect(component).toMatchSnapshot();
    });

    // Not really needed covered in snapshot testing
    it('should have the text label ', () => {
        const component = shallow(<Button text={'test label'} />);
        const text = component.find('span').text();
        expect(text).toEqual('test label');
    });
});

describe('Button onClick event testing ', () => {
    it(' Check if onClick function is getting called ', () => {
        const onClickEventFunc = jest.fn();
        const component = mount(<Button onClick={onClickEventFunc} />); // if we would have used shallow then also it would have worked as we are not using any child component
        component.simulate('click');
        expect(onClickEventFunc).toBeCalled();
    });
});
