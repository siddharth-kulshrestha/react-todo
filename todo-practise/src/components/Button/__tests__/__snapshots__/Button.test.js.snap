// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button  should render correctly  1`] = `
<span
  className="button-container"
>
  hello world
</span>
`;
